### PR TITLE
Add filter and search

### PR DESCRIPTION
**Pull request by larsonkonr**

### CODE REVIEW

File: app/assets/javascripts/jobs.js [hunk 1]
```diff
@@ -39,12 +39,12 @@ function likeJob(id, button) {
...
+  buttons.parent("div").append("<h5 class='added-job'>Added to My Jobs</h5>");
   buttons.remove();
 }
 
 function addedToHiddenJobs(button) {
-  $(button).parent("div").parent("td").parent("tr").remove();
+  $(button).parent("div").parent("div").parent("li").remove();
   $(".flash-container").show().addClass("alert-success")
     .text("Successfully Removed");
   setTimeout(function(){
```
Comment (scottcrawford03) [inline app/assets/javascripts/jobs.js:50]: nice work on this

File: app/assets/javascripts/sidebar.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+$(document).ready(function() {
+  function initializeSearchBar() {
+    var tags = ["hello"];
```
Comment (scottcrawford03) [inline app/assets/javascripts/sidebar.js:3]: what is this right here?

File: app/assets/javascripts/sidebar.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      $(".search-item").hide().filter(function() {
+        var itemText = $(this).text().toLowerCase();
+        var searchText = $("#search-bar").val().toLowerCase();
+        tags.push(searchText);
+        var filterMatch = itemText.indexOf(searchText) != -1;
+        return filterMatch;
+      }).show();
+    });
+  }
+
```
Comment (scottcrawford03) [inline app/assets/javascripts/sidebar.js:15]: i like it.

File: app/controllers/my_jobs_controller.rb [hunk 4]
```diff
@@ -1,5 +1,6 @@
 class MyJobsController < ApplicationController
   def index
     @jobs = current_user.jobs_liked.includes(:company)
+    @tags = current_user.jobs_liked.map { |j| j.tags.map(&:name) }.flatten.uniq
```
Comment (scottcrawford03) [inline app/controllers/my_jobs_controller.rb]: we could use .flat_map here then you don't have to call flatten at the end

File: app/controllers/jobs_controller.rb [hunk 5]
```diff
@@ -3,11 +3,10 @@ class JobsController < ApplicationController
 
   def index
     @jobs = Job.visible_for(current_user).includes(:company)
+    @tags = Tag.pluck(:name)
```
Comment (bryceholcomb) [inline app/controllers/jobs_controller.rb:6]: Looks good, no need to instantiate the tags

**Comment by bryceholcomb** [comment]
Looks good, almost there with the base functionality!

**Pull request merged**